src/vanilla/ClientModelExtensions.cs

```diff
@@ -630,314 +671,246 @@ public static string ConstructMapper(this IModelType type, string serializedName
                 constraints = parameter.Constraints;
             }
 
-            builder.AppendLine("").Indent();
-
-            bool wroteXmlName = isXML && !string.IsNullOrEmpty(xmlName) && xmlName != serializedName;
-            if (wroteXmlName)
-            {
-                builder.AppendLine($"xmlName: '{xmlName}',");
-            }
-
-            if (!isCaseSensitive)
+            value.Object(mapper =>
             {
-                serializedName = serializedName.ToLower();
-            }
+                bool wroteXmlName = isXML && !string.IsNullOrEmpty(xmlName) && xmlName != serializedName;
```

Comment:
should this be called `shouldWriteXmlName` or similar?

Reply:
Changed to `addXmlNameParameterValue`. This variable really keeps track of whether we should add the xmlName parameter that is passed to this function or whether we should use the xmlName property on the property variable.